MapboxNavigation/BottomBannerViewController.swift

```diff
@@ -62,11 +68,20 @@ open class BottomBannerView: UIView, NavigationComponent {
         removeTimer()
     }
     
-    override open func willMove(toSuperview newSuperview: UIView?) {
-        super.willMove(toSuperview: newSuperview)
-        if superview != nil, newSuperview == nil {
-            removeTimer()
-        }
+    override open func loadView() {
+        let root: BottomBannerView = .forAutoLayout() //Must use local var to prevent generic factory from messing up.
```

Comment:
Huh, interesting. Out of curiosity, how does it mess up?

Reply:
Somehow, type inference initializes a `UIView` if you do
```swift
let view = BottomBannerView.forAutoLayout()
```

This can be fixed by specifiying the generic type as an argument in the generic function.

Will be interesting to see how Swift 5 handles this.

Reply:
Hm, I guess that behavior sounds reasonable actually. I’d expect `.forAutoLayout<BottomBannerView>()` to be required in that case.

Reply:
I don't think action is needed here. Clarity at point-of-site is more important than brevity, and here it's very apparent that the root view is a `BottomBannerView`.